### PR TITLE
Add a failing test for `state.can` that involves parallel regions

### PR DESCRIPTION
**Pull request by Andarist**
Add a failing test for regression from https://github.com/statelyai/xstate/pull/2804

### CODE REVIEW

**Comment by davidkpiano** [comment]
Please see https://github.com/statelyai/xstate/pull/2839 for a potential fix, but also one that reveals a bigger problem: not all targets appear in the transition data.

IMO this is a much smaller edge-case than what #2804 fixed, and should not prevent #2804 from going in.

File: packages/core/test/state.test.ts [hunk 1]
```diff
@@ -846,6 +846,42 @@ describe('State', () => {
...
+        type: 'parallel',
+        states: {
+          a: {
+            initial: 'a1',
+            states: {
+              a1: {
+                id: 'foo',
+                on: {
+                  // first region doesn't change value here
+                  EVENT: ['#foo', '#bar']
```
Comment (davidkpiano) [inline packages/core/test/state.test.ts:862]: It should be this:

```suggestion
                  EVENT: { target: ['#foo', '#bar'] }
```

**Comment by Andarist** [comment]
closing this as it became a part of the https://github.com/statelyai/xstate/pull/2839
